products/billingbudget/api.yaml

```diff
@@ -172,16 +172,22 @@ objects:
             properties:
               - !ruby/object:Api::Type::String
                 name: pubsubTopic
-                required: true
                 description: |
                   The name of the Cloud Pub/Sub topic where budget related
                   messages will be published, in the form
                   projects/{project_id}/topics/{topic_id}. Updates are sent
                   at regular intervals to the topic.
               - !ruby/object:Api::Type::String
                 name: schemaVersion
-                default_value: "1.0"
```

Comment:
The description states only "1.0" is accepted. Why do we want to remove this as default?

Reply:
From [the API reference doco for the `allUpdatesRule`](https://cloud.google.com/billing/docs/reference/budget/rest/v1beta1/billingAccounts.budgets#allupdatesrule) and my usage of the API directly the `schemaVersion` field only applies to `pubsubTopic`, and doesn't apply to `monitoringNotificationChannels`.

For example using [`billingAccounts.budgets.create`](https://cloud.google.com/billing/docs/reference/budget/rest/v1beta1/billingAccounts.budgets/create) you can provide both `schemaVersion` and `monitoringNotificationChannels` (and not `pubsubTopic`), and the object returned after successful creation of the resource does not include `schemaVersion` so it seems to be ignored/discarded (examples below).

So I removed this line as I thought it'd be more correct to have the configuration stored in Terraform state match what's configured in the API.
But can see removing this will cause issues for existing users of this resource if they aren't explicitly specifying the `schemaVersion` value and are relying on the default value.

I know this is a beta API, so I'm wondering if it would make more sense if the `schemaVersion` was a child of the `pubsubTopic`. But I obviously don't know what plans are in-store for future features in the API, so I could be missing something and there be good reason for this.


Example - Provided budget:
```json
{
  "budget": {
    "allUpdatesRule": {
      "monitoringNotificationChannels": [
        "projects/{project_id}/notificationChannels/{channel_id}"
      ],
      "schemaVersion": "1.0"
    },
    "displayName": "Budget display name",
    "amount": {
      "specifiedAmount": {
        "currencyCode": "USD",
        "units": 100
      }
    },
    "thresholdRules": [
      {
        "thresholdPercent": 1.0,
        "spendBasis": "CURRENT_SPEND"
      }
    ],
    "budgetFilter": {
      "projects": [
        "projects/{project_id}"
      ]
    }
  }
}
```

Example - Returned budget:
```json
{
  "name": "billingAccounts/{billingAccountId}/budgets/{budgetId}",
  "displayName": "Budget display name",
  "budgetFilter": {
    "projects": [
      "projects/{project_number}"
    ],
    "creditTypesTreatment": "INCLUDE_ALL_CREDITS"
  },
  "amount": {
    "specifiedAmount": {
      "currencyCode": "USD",
      "units": "100"
    }
  },
  "thresholdRules": [
    {
      "thresholdPercent": 1,
      "spendBasis": "CURRENT_SPEND"
    }
  ],
  "allUpdatesRule": {
    "monitoringNotificationChannels": [
      "projects/{project_id}/notificationChannels/{channel_id}"
    ]
  },
  "etag": "{etag_id}"
}
```

Reply:
Tested this a bit myself and it seems like it wouldn't hurt to keep the `default_value` for Schema version. Like you've observed it gets discarded by the API but doesn't cause a diff in Terraform. I think leaving it in would be the easiest way to ensure we don't bother existing users.

Reply:
Done! Added this back